### DISCUSSION TITLE
Listening for events from external process

### DISCUSSION THREAD

**Discussion by bokolob** [miguelgrinberg/python-socketio#720] [Q&A]
Hello,
I have a specific problem. I need not only to emit events but listen to them too.  Kind of a bot that joins some rooms and gets all the messages from there.
On the one hand, I can use the client library, but I would prefer to use just bare redis for both emitting and listening, as not to overengineer my system. 

Is there a way to do it?

**Comment by miguelgrinberg** [comment]
Would it make sense for your bot to join the room as a client instead of a server? There is a Socket.IO client in this package in addition to the server. Check the docs for instructions.

**Comment by bokolob** [comment]
Yes, I know about the client library, but then I will need to maintain an HTTP/WebSocket connection and authorization, but I just need some reduced server functionality. Here is the possibility to emit events through Redis, I wonder if there is a similar symmetric API  for getting them.

**Comment by miguelgrinberg** [comment]
No, that is not something that this package supports. I don't see how this is "reduced" functionality. If you need to emit and to receive events than that's exactly what the server does. You can consider moving the bot to your server process, or else as I said, make it a client.
